### ISSUE TITLE
[BUG] Blank Password in proxy results in no credential sent for authentication

### ISSUE THREAD

**Issue by jcvillalta03** [microsoft/playwright-java#152]
There are 3rd party proxy services like Crawlera, that require its users to use a blank password (empty string), for the its service. Please see the the [Proxy API](https://doc.scrapinghub.com/crawlera-proxy-api.html#proxy-api) for reference.

I am starting to look into the cause of this. It may be due to some default GSON config that treats blank strings as nulls when converting to JSON.

Here is a sample test (adapted from TestBrowserContextProxy) that demonstrates the failure:

```
  @Test
  void shouldAuthenticateWithBlankPassword() {
    server.setRoute("/target.html", exchange -> {
      List<String> auth = exchange.getRequestHeaders().get("proxy-authorization");
      if (auth == null) {
        exchange.getResponseHeaders().add("Proxy-Authenticate", "Basic realm='Access to internal site'");
        exchange.sendResponseHeaders(407, 0);
        exchange.getResponseBody().close();
        return;
      }
      exchange.sendResponseHeaders(200, 0);
      try (OutputStreamWriter writer = new OutputStreamWriter(exchange.getResponseBody())) {
        writer.write("<html><title>" + auth.get(0) + "</title></html>");
      }
    });
    BrowserContext context = browser.newContext(new Browser.NewContextOptions().setProxy()
      .withServer("localhost:" + server.PORT)
      .withUsername("user")
      .withPassword("").done());
    Page page = context.newPage();
    page.navigate("http://non-existent.com/target.html");
    assertEquals("Basic " + Base64.getEncoder().encodeToString("user:".getBytes()), page.title());
    context.close();
  }
```

The assertion fails because the `page.title()` evaluates to "" (empty string)

**Comment by yury-s** [comment]
This turns out to be a bug in Chromium client upstream, should be fixed by https://github.com/microsoft/playwright/pull/4779

**Comment by yury-s** [comment]
The fix is now available in 0.180.0-SNAPSHOT
